buildSrc/src/main/kotlin/TaskOptionExtractor.kt

```diff
@@ -0,0 +1,92 @@
+import org.gradle.api.tasks.options.Option
+import java.lang.reflect.Method
+
+/**
+ * Utility class for extracting task options from Gradle task classes via reflection.
+ */
+object TaskOptionExtractor {
+
+    /**
+     * Extracts task options from one or more Gradle task classes via reflection.
+     * Looks for methods annotated with @Option and extracts option names, descriptions, and possible values.
+     *
+     * @param classNames List of fully qualified class names to extract options from (e.g., task class and its parents)
+     * @return List of extracted options
+     */
+    fun extractTaskOptions(vararg classNames: String): List<TaskOptionDescriptor> {
+        if (classNames.isEmpty()) {
+            throw RuntimeException("no expected class names provided")
+        }
+
+        val classes = classNames.map { Class.forName(it) }
+
+        val allMethods = classes.flatMap { it.declaredMethods.toList() }
+
+        return allMethods.mapNotNull { method ->
+            val optionAnnotation = method.getAnnotation(Option::class.java)
+            if (optionAnnotation != null) {
+                TaskOptionDescriptor(
+                    optionName = optionAnnotation.option,
+                    description = optionAnnotation.description,
+                    possibleValues = extractPossibleValuesFromEnumParameters(method),
+                    requiresArgument = !isBooleanOption(method)
+                )
+            } else {
+                null
+            }
+        }.distinctBy { it.optionName }.sortedBy { it.optionName }
+    }
```

Comment:
💅 Let's use Kotlin idioms for requirements, collection processing and write this function as an expression:
```suggestion
    fun extractTaskOptions(vararg classNames: String): List<TaskOptionDescriptor> =
        classNames.also { require(it.isNotEmpty()) { "no expected class names provided" } }
            .map { Class.forName(it) }
            .flatMap { it.declaredMethods.toList() }
            .mapNotNull { method ->
                method.getAnnotation(Option::class.java)
                    ?.let { annotation ->
                        TaskOptionDescriptor(
                            optionName = annotation.option,
                            description = annotation.description,
                            possibleValues = extractPossibleValuesFromEnumParameters(method),
                            requiresArgument = !isBooleanOption(method)
                        )
                    }
            }
            .distinctBy { it.optionName }
            .sortedBy { it.optionName }
```